packages/react/src/components/Button/SplitButton/SplitButton.styles.ts

```diff
@@ -27,92 +27,128 @@ export const getStyles = memoizeFunction((theme: ITheme, customStyles?: IButtonS
       getFocusStyle(theme, { highContrastStyle: buttonHighContrastFocus, inset: 2, pointerEvents: 'none' }),
       {
         display: 'inline-flex',
-        selectors: {
-          '.ms-Button--default': {
-            borderTopRightRadius: '0',
-            borderBottomRightRadius: '0',
-            borderRight: 'none',
-            flexGrow: '1',
-          },
-          '.ms-Button--primary': {
-            borderTopRightRadius: '0',
-            borderBottomRightRadius: '0',
-            border: 'none',
-            flexGrow: '1',
+        '.ms-Button--default': {
+          borderTopRightRadius: '0',
+          borderBottomRightRadius: '0',
+          borderRight: 'none',
+          flexGrow: '1',
+        },
+        '.ms-Button--primary': {
+          borderTopRightRadius: '0',
+          borderBottomRightRadius: '0',
+          border: 'none',
+          flexGrow: '1',
 
-            selectors: {
-              [HighContrastSelector]: {
-                color: 'WindowText',
-                backgroundColor: 'Window',
-                border: '1px solid WindowText',
-                borderRightWidth: '0',
-                ...getHighContrastNoAdjustStyle(),
-              },
-              ':hover': {
-                border: 'none',
-              },
-              ':active': {
-                border: 'none',
+          [HighContrastSelector]: {
+            color: 'WindowText',
+            backgroundColor: 'Window',
+            border: '1px solid WindowText',
+            borderRightWidth: '0',
+            ...getHighContrastNoAdjustStyle(),
+          },
+          ':hover': {
+            backgroundColor: 'Highlight',
+            borderColor: 'Highlight',
+            borderRightWidth: '0',
+            color: 'Window',
+          },
+          ':active': {
+            borderColor: 'Highlight',
+          },
+        },
+        '.ms-Button--default + .ms-Button': {
+          [HighContrastSelector]: {
+            border: '1px solid WindowText',
+            borderLeftWidth: '0',
+            ':hover': {
+              borderColor: 'Highlight',
+              color: 'Highlight',
```

Comment:
I'd suggest also adding `backgroundColor: HighlightText` here too, to reduce the chances of some other style breaking the text/bg color pairing at a different specificity.